scripts/gas-refund-program/transactions-indexing/swaps-subgraph.ts

```diff
@@ -74,17 +70,23 @@ export async function getSwapsForAccounts({
 
     const swaps = data.data.swaps;
 
-    assert(swaps.length !== 1000, 'unsafe fix pagination');
-
     return swaps;
   };
 
-  // array of sliced results, without slicing breaks with Payload too large (too many `initiators`)
-  const result = await Promise.all(
-    sliceCalls({ inputArray: accounts, execute, sliceLength: 1000 }),
-  );
+  let swaps: SwapData[] = [];
+  let skip = 0;
+  let pageSize = 100;
+
+  while (true) {
+    const _swaps = await query(skip, pageSize);
+    swaps = swaps.concat(_swaps);
+    if (_swaps.length < pageSize) {
+      break;
+    }
+    skip = skip + pageSize;
```

Comment:
Does pagination always run ok? I seem to remember there were some problems with Graph pagination...

Reply:
It's true that thegraph pagination is not reliable at some point. But this is only a secondary loop.
<img width="555" alt="Screenshot 2022-05-02 at 12 48 03" src="https://user-images.githubusercontent.com/5844302/166222710-61b2d307-e14d-4cdc-ac31-d641db363149.png">

The primary loop is sliding a 6h window [here](https://github.com/paraswap/paraswap-volume-tracker/blob/cf913fdfb8675b8ed7e31aa6a47a2ea59f819b75/scripts/gas-refund-program/transactions-indexing/successful-swap-tx-indexing.ts#L62). 

The secondary (this one), scans 100 transactions in batch.

The assumption is that we won't scan more than 5000 tx within 6h. If this goes wrong, first the script will crash. We can always lower the primary loop cycle.

I didn't see it crash with our conditions though + we're migrating to another source soon or later. I'm open to suggestion

Reply:
I can confirm `swaps` array ends up with some duplications in it. Not yet sure whether it's harmful or not, didn't yet follow the logic further